crates/libs/windows/src/core/factory_cache.rs

```diff
@@ -118,6 +119,29 @@ pub fn factory<C: RuntimeName, I: Interface>() -> Result<I> {
     }
 }
 
+/// Attempts to load the factory interface for the given WinRT class located
+/// in the specified library.
+pub fn factory_from_library<C: RuntimeName, I: Interface>(library: &str) -> Result<I> {
+    let name = HSTRING::from(C::NAME);
+    unsafe {
+        let abi = get_activation_factory(library.as_bytes(), &name)?;
```

Comment:
Doesn't the library name have to be null terminated? Rust strings are not null terminated.

Reply:
Yes, the underlying `delay_load` function assumes the byte array is null terminated. The `factory_cache` (in master) ensures that the library name is terminated.